### ISSUE TITLE
Cross Platform Firmware Build Scripts

### ISSUE THREAD

**Issue by korczis** [brandonlw/Psychson#15]
It will be really nice if the build scripts will work on different OS than Win.

Quick and dirty prototype can be found here - https://github.com/korczis/Psychson/blob/master/firmware/build.rb 

If there will be demand I can completely rewrite all build scripts to ruby.

**Comment by necrose99** [comment]
well mono is an option.....

**Comment by korczis** [comment]
@necrose99 You are wrong.

Mono is option for compilation of csharp ... Not an option for replacement of .bat scripts.

Anyway, I checked the csharp tools and I am scared it is not portable using mono. What about win specific access to devices via OpenFile?

Do you think this will run under the mono?

https://github.com/adamcaudill/Psychson/blob/master/DriveCom/DriveCom/PhisonDevice.cs#L134

I bet it wont.

**Comment by MrSchism** [comment]
**_TL;DR:**_ _I agree with the ruby option, but only if we go with a OS-agnostic scripts._

There are really two options here:
1. OS-specific scripts.  They'd take up more space as a whole, but the scripts for the individual OS would be smaller... and would probably work better individually.  Biggest down side is that they'd have to be maintained separately and possibly in different languages.
2.  OS-agnostic scripts.  They'd take up less space overall, but would likely contain bloat for compatibility.

If we went with OS-agnostic scripts, ruby or python would probably be the way to go.  Java could work, too....but I'm definitely not a fan of java.

**Comment by necrose99** [comment]
i found a CMD to shell on Github and vice versa 
https://github.com/BYVoid/Batsh
few others simular... 
https://github.com/ea/batch2bash

some others thad have dual roud tripiing on the gagillion usefull sys-min toys i've stared of late... 

I just mentioed mono as a posible future option...  as one could in theory have this runing quite nicely on many platforms..... it dosnt have to be mono it was a thought....  and at worst thiers a C#/C++ round triper on codeplex,  python and python-C++ or a few other posibles... i just see quite a bit of Promiss 

but what do I know , I just am hoping to package this the moment it's a Linux usable... :-)

**Comment by raif7** [comment]
I second the Linux usable motion :-)
